### PR TITLE
jetbrains: add mainPrograms

### PR DESCRIPTION
**Pull request by Atemu**
###### Motivation for this change

I need to unset DESKTOP_STARTUP_ID on jetbrain IDE startup to make @fctorial's https://github.com/fctorial/enter_nix_shell work to its full potential. In order to [do the wrapping in overrides generically](https://github.com/Atemu/nixos-config/blob/06be43876e0a81d4a283b287bbe1f2098d5f129e/overlays.nix#L23-L31), I needed the IDEs' executable names.

Also, did some cleanup. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
I have no idea why ofBorg fails to find the attr, `nix eval -f . jetbrains.jdk.meta.mainProgram` works just fine on my checkout.

**Comment by SuperSandro2000** [comment]
@Atemu please fix the eval error.

**Comment by Atemu** [comment]
The problem is, I can't reproduce the eval error. Are you able to?

**Comment by SuperSandro2000** [comment]
> The problem is, I can't reproduce the eval error. Are you able to?

I think meta.mainProgram is missing here https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/openjdk/darwin/11.nix#L73 when evaling on darwin.

**Comment by Atemu** [comment]
Ahh, Darwin's openjdks' meta were basically non-existent; fixed that.

File: pkgs/development/compilers/openjdk/meta.nix [hunk 1]
```diff
@@ -0,0 +1,8 @@
+lib: with lib; {
+  homepage = "http://openjdk.java.net/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/openjdk/meta.nix:2]:
```suggestion
  homepage = "https://openjdk.java.net/";
```

**Pull request merged**